Mathlib/Analysis/Complex/Hadamard.lean

```diff
@@ -0,0 +1,419 @@
+    simpa [F, abs_invInterpStrip _ _ hε, norm_smul, mul_comm] using
+      norm_mul_invInterpStrip_le_one_of_mem_verticalClosedStrip f ε hε z hd hB hz
+  · simp only [Real.rpow_pos_of_pos (sSupNormIm_eps_pos f hε _) z.re]
+  · simp only [Real.rpow_pos_of_pos (sSupNormIm_eps_pos f hε _) (1-z.re)]
+
+lemma eventuallyle (z : ℂ) (hB : BddAbove ((norm ∘ f) '' (verticalClosedStrip 0 1)))
+    (hd : DiffContOnCl ℂ f (verticalStrip 0 1)) (hz : z ∈ verticalStrip 0 1) :
+    (fun _ : ℝ ↦ ‖f z‖) ≤ᶠ[𝓝[>] 0]
+    (fun ε ↦ ‖((ε + sSupNormIm f 0) ^ (1 - z) * (ε + sSupNormIm f 1) ^ z : ℂ)‖) := by
+  filter_upwards [self_mem_nhdsWithin] with ε (hε : 0 < ε) using
+    norm_le_interpStrip_of_mem_verticalClosedStrip_eps f ε hε z hB hd
+      (mem_of_mem_of_subset hz (preimage_mono Ioo_subset_Icc_self))
+
+lemma norm_le_interpStrip_of_mem_verticalStrip_zero (z : ℂ)
+    (hd : DiffContOnCl ℂ f (verticalStrip 0 1))
+    (hB : BddAbove ((norm ∘ f) '' (verticalClosedStrip 0 1))) (hz : z ∈ verticalStrip 0 1) :
+    ‖f z‖ ≤ ‖interpStrip f z‖ := by
+  apply tendsto_le_of_eventuallyLE _ _ (eventuallyle f z hB hd hz)
+  · apply tendsto_inf_left
+    simp only [tendsto_const_nhds_iff]
+  -- Proof that we can let epsilon tend to zero.
+  · rw [interpStrip_eq_of_mem_verticalStrip _ _ hz]
+    convert ContinuousWithinAt.tendsto _ using 2
+    · simp only [ofReal_zero, zero_add]
+    · simp_rw [← ofReal_add, norm_eq_abs]
+      have : ∀ x ∈ Ioi 0, (x + sSupNormIm f 0) ^ (1 - z.re) * (x + (sSupNormIm f 1)) ^ z.re
+          = abs (↑(x + sSupNormIm f 0) ^ (1 - z) * ↑(x + sSupNormIm f 1) ^ z) := by
+              intro x hx
+              simp only [map_mul]
+              repeat rw [abs_cpow_eq_rpow_re_of_nonneg (le_of_lt (sSupNormIm_eps_pos f hx _)) _]
+              · simp only [sub_re, one_re]
+              · simpa using (ne_comm.mpr (ne_of_lt hz.1))
+              · simpa [sub_eq_zero] using (ne_comm.mpr (ne_of_lt hz.2))
+      apply tendsto_nhdsWithin_congr this _
+      simp only [zero_add]
+      rw [map_mul, abs_cpow_eq_rpow_re_of_nonneg (sSupNormIm_nonneg _ _) _,
+        abs_cpow_eq_rpow_re_of_nonneg (sSupNormIm_nonneg _ _) _]
+      · apply Tendsto.mul
+        · apply Tendsto.rpow_const
+          · nth_rw 2 [← zero_add (sSupNormIm f 0)]
+            exact Tendsto.add_const (sSupNormIm f 0) (tendsto_nhdsWithin_of_tendsto_nhds
+              (Continuous.tendsto continuous_id' _))
+          · right; simp only [sub_nonneg, le_of_lt hz.2]
+        · apply Tendsto.rpow_const
+          · nth_rw 2 [← zero_add (sSupNormIm f 1)]
+            exact Tendsto.add_const (sSupNormIm f 1) (tendsto_nhdsWithin_of_tendsto_nhds
+              (Continuous.tendsto continuous_id' _))
+          · right; simp only [sub_nonneg, le_of_lt hz.1]
+      · simpa using (ne_comm.mpr (ne_of_lt hz.1))
+      · simpa [sub_eq_zero] using (ne_comm.mpr (ne_of_lt hz.2))
+
+/--
+**Hadamard three-line theorem** on `[0,1]`: If `f` is a bounded function, continuous on `[0,1]`
```

Comment:
The docstring looks like `f` is a function on `[0, 1]`. But instead it's a function on the vertical strip. Could you clarify the docstring of this theorem and of the next one?

Reply:
Done :)